pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,125 @@
+
+### Schema Registry validation
+Producers and consumers do not communicate with each other directly; rather, information transfer happens via Kafka. At the same time, the consumer still needs to know the type of data the producer is sending. Imagine if the producer starts sending bad data to Kafka or if the data type of your data gets changed. We need a way to have a common data type that must be agreed upon.
+
+This is where Schema Registry comes into play. It is an application that runs outside your Kafka protocol and handles schema distribution to producers and consumers by storing a copy of the schema in its local cache and validating them in Kafka.
+
+```mermaid
+sequenceDiagram
+Producer ->> Schema Registry: Validate message schema
+Schema Registry ->> Producer: Return validation result
+
+Producer ->> Kafka: Publish message
+Kafka ->> Consumer: Recieves message
+Consumer ->> Schema Registry: Validate message schema
+Schema Registry ->> Consumer: Return validation result
+```
+
+With the Schema Registry in place, the producer first talks to the Schema Registry and checks if the schema is available before sending the data to Kafka. If it cannot locate the schema, it registers and caches it in the Schema Registry. When the producer receives the schema, it sends it to Kafka prefixed with a unique schema ID. When the consumer processes this message, it will communicate with the Schema Registry using the schema ID obtained from the producer. If there is a schema mismatch, the Schema Registry will throw an error, informing the producer that it violates the schema agreement.
+
+```mermaid
+stateDiagram-v2
+    Producer --> Broker : messages
+    Producer --> Fail : error
+    Broker --> Schema_registry : validation request
+    Schema_registry --> consumer : valid message
+    Schema_registry --> Fail : invalid message
+```
+
+AsyncAPI is not directly involved in validation based on the Schema Registry. The good thing is that you do not have to duplicate schemas in your AsyncAPI document stored in Schema Registry. You can reference schemas from Schema Registry in your AsyncAPI documents.
+Here's an example of an AsyncAPI document where you can see both `schemaFormat` and `payload` referenced from the Schema Registry:
+```yml
+asyncapi: 2.5.0
+info:
+  title: Example with Avro
+  version: 0.1.0
+channels:
+  example:
+    publish:
+      message:
+        schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
+        payload:
+          $ref: 'https://example.europe-west3.gcp.confluent.cloud/subjects/test/versions/1/schema'
+```
+
+---
+
+<DocsButton
+  suggestions={[
+    {
+      href: '/docs/guides/validate',
+      title: 'Validate AsyncAPI Documents',
+      type:'back',
+    },
+    {
+      href: '/docs/reference',
+      title: 'Reference - Overview',
+      type:'next',
+    }
+  ]}
+/>
```

Comment:
```suggestion
```

@starlightknown Can you please remove this from Doc page as we got automation regarding Prev and Next Button?

Reply:
thank you for helping spread the word of this change @akshatnema, appreciate the help 😄

Reply:
shall I remove it for all?

Reply:
Yes @starlightknown, go ahead and remove your previous/next buttons in all of your PRs because this new improvement @akshatnema  made so buttons will now be automated 🌟